futures-executor/src/local_pool.rs

```diff
@@ -0,0 +1,167 @@
+//! TODO: dox
+
+use std::prelude::v1::*;
+
+use std::cell::{RefCell};
+use std::rc::{Rc, Weak};
+
+use futures_core::{Future, Poll, Async, Stream};
+use futures_core::task::{Context, Waker, LocalMap};
+use futures_core::executor::{Executor, SpawnError};
+use futures_util::stream::FuturesUnordered;
+
+use thread::ThreadNotify;
+use enter;
+
+struct Task {
+    fut: Box<Future<Item = (), Error = ()>>,
+    map: LocalMap,
+}
+
+/// todo: dox
+pub struct LocalPool {
+    pool: FuturesUnordered<Task>,
+    incoming: Rc<Incoming>,
+}
+
+/// todo: dox
+pub struct LocalExecutor {
+    incoming: Weak<Incoming>,
+}
+
+type Incoming = RefCell<Vec<Task>>;
+
+fn run_executor<T, F: FnMut(&Waker) -> Async<T>>(mut f: F) -> T {
+    let _enter = enter()
+        .expect("cannot execute `LocalPool` executor from within \
```

Comment:
Is this still necessary?

Reply:
Yes -- I think we still want `enter` to ensure that executors aren't accidentally nested.